pkg/restore/restore.go

```diff
@@ -917,44 +924,59 @@ func (r *pvRestorer) executePVAction(obj *unstructured.Unstructured) (*unstructu
 		return obj, nil
 	}
 
-	// If we can't find a snapshot record for this particular PV, it most likely wasn't a PV that Ark
-	// could snapshot, so return early instead of trying to restore from a snapshot.
-	backupInfo, found := r.volumeBackups[pvName]
-	if !found {
-		return obj, nil
-	}
-
-	// Past this point, we expect to be doing a restore
-
-	if r.blockStore == nil {
-		return nil, errors.New("you must configure a persistentVolumeProvider to restore PersistentVolumes from snapshots")
-	}
-
-	log := r.logger.WithFields(
-		logrus.Fields{
-			"persistentVolume": pvName,
-			"snapshot":         backupInfo.SnapshotID,
-		},
-	)
-
-	log.Info("restoring persistent volume from snapshot")
-	volumeID, err := r.blockStore.CreateVolumeFromSnapshot(backupInfo.SnapshotID, backupInfo.Type, backupInfo.AvailabilityZone, backupInfo.Iops)
+	volumeSnapshots, err := r.backupStore.GetBackupVolumeSnapshots(r.backupName)
 	if err != nil {
 		return nil, err
 	}
-	log.Info("successfully restored persistent volume from snapshot")
 
-	updated1, err := r.blockStore.SetVolumeID(obj, volumeID)
-	if err != nil {
-		return nil, err
+	for _, snapshot := range volumeSnapshots {
+		if snapshot.Spec.ProviderVolumeID == pvName {
```

Comment:
these two things aren't expected to be equal (ProviderVolumeID is e.g. the AWS EBS volume ID in AWS, while pvName is the name of the PV object in Kubernetes), looks like we probably need to add `PersistentVolumeName` to the `VolumeSnapshot` struct